database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -85,8 +85,32 @@ public String parseSnapshot(DataSnapshot snapshot) {
                 return snapshot.getKey();
             }
         });
+    }
+
+    @Override
+    public ChangeEventListener addChangeEventListener(@NonNull ChangeEventListener listener) {
+        boolean wasListening = isListening();
+        super.addChangeEventListener(listener);
+
+        // Only start listening when the first listener is added
+        if (!wasListening) {
```

Comment:
This is obviously a good fix.  It looks pretty similar to the code we have in `FirebaseArray` though, can we add a TODO to investigate if we should push some of this logic up into a base class?